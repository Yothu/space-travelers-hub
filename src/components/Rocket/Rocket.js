import styled from 'styled-components';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket, cancelRocket } from '../../redux/rockets/rockets';
import '../Font.css';

const RocketContainer = styled.div`
  display: flex;
  padding: 0.7rem 0;
`;

const DataContainer = styled.div`
  padding: 0 1rem;
`;

const Name = styled.h3`
  margin: 0;
  font-size: 1rem;
  font-weight: 500;
  font-family: Arial, Helvetica, sans-serif !important;
`;

const Description = styled.p`
  font-size: 0.7rem;
`;

const ImageStyle = {
  width: '12rem',
  height: '9rem',
};

const ReserveBtn = styled.button`
  border: none;
  color: white;
  cursor: pointer;
  padding: 0.5rem;
  font-size: 0.7rem;
  border-radius: 3px;
  background-color: #4844FF;
`;

const CancelBtn = styled.button`
  border: 1px solid gray;
  color: gray;
  cursor: pointer;
  padding: 0.5rem;
  font-size: 0.7rem;
  border-radius: 3px;
  background-color: white;
`;

const ReservedTag = styled.span`
  color: white;
  padding: 0.1rem 0.2rem;
  font-size: 0.5rem;
  font-weight: bold;
  border-radius: 3px;
  margin-right: 0.4rem;
  background-color: #00a2b9;
`;

const Rocket = ({
  id, image, name, description, reserved,
}) => {
  const dispatch = useDispatch();
  const Image = () => <img src={image} style={ImageStyle} alt={name} />;

  const resRocket = () => {
    dispatch(reserveRocket(id));
  };

  const cancelReservation = () => {
    dispatch(cancelRocket(id));
  };

  return (
    <RocketContainer id={id}>
      <Image />
      <DataContainer>
        <Name className="font-1">{name}</Name>
        <Description className="font-1">
          {reserved === true && (
            <ReservedTag>Reserved</ReservedTag>
          )}
          {description}
        </Description>
        {reserved === true && (
          <CancelBtn className="font-1" onClick={cancelReservation}>Cancel Reservation</CancelBtn>
        )}
        {reserved === false && (
          <ReserveBtn className="font-1" onClick={resRocket}>Reserve Rocket</ReserveBtn>
        )}
      </DataContainer>
    </RocketContainer>
  );
};

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  description: PropTypes.string.isRequired,
};

export default Rocket;
