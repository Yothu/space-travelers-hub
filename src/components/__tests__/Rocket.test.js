import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rocket from '../Rocket/Rocket';
import store from '../../redux/configureStore';

describe('Rocket component', () => {
  it('it renders', () => {
    const mock = {
      id: 2,
      image: 'image',
      name: 'Falcon 9',
      reserved: false,
      description: 'very rocket',
    };

    const rocket = renderer
      .create(
        <Provider store={store}>
          <Rocket
            id={mock.id}
            key={mock.id}
            name={mock.name}
            image={mock.image}
            reserved={mock.reserved}
            description={mock.description}
          />
        </Provider>,
      )
      .toJSON();
    expect(rocket).toMatchSnapshot();
  });
});
