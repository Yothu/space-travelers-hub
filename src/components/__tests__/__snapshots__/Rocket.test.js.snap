// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component it renders 1`] = `
<div
  className="sc-bdvvtL kbYwEE"
  id={2}
>
  <img
    alt="Falcon 9"
    src="image"
    style={
      Object {
        "height": "9rem",
        "width": "12rem",
      }
    }
  />
  <div
    className="sc-gsDKAQ iLjpKD"
  >
    <h3
      className="sc-dkPtRN YpXkX font-1"
    >
      Falcon 9
    </h3>
    <p
      className="sc-hKwDye iiXaFm font-1"
    >
      very rocket
    </p>
    <button
      className="sc-eCImPb dMdLSA font-1"
      onClick={[Function]}
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
